library/agent/hooks/VersionedPackage.ts

```diff
@@ -15,25 +14,57 @@ export class VersionedPackage {
     return this.range;
   }
 
-  addFile(relativePath: string): WrappableFile {
-    const file = new WrappableFile(relativePath);
-    this.files.push(file);
-
-    return file;
+  onRequire(interceptor: RequireInterceptor) {
+    if (typeof interceptor !== "function") {
+      throw new Error("Interceptor must be a function");
+    }
+    this.requireInterceptors.push(interceptor);
+    // Allow chaining
+    return this;
   }
 
-  addSubject(selector: (exports: any) => unknown): WrappableSubject {
-    const fn = new WrappableSubject(selector);
-    this.subjects.push(fn);
+  onFileRequire(relativePath: string, interceptor: RequireInterceptor) {
+    if (typeof relativePath !== "string") {
+      throw new Error("Relative path must be a string");
+    }
+    if (typeof interceptor !== "function") {
+      throw new Error("Interceptor must be a function");
+    }
```

Comment:
Doesn't TypeScript enforce this? Same for relative path? or does this come from externally?